test/bitarray.jl

```diff
@@ -203,6 +203,15 @@ timesofar("utils")
         @test_throws DimensionMismatch BitMatrix((isodd(i) for i in 1:3))
     end
 
+    @testset "constructor from infinite iterable" begin
+        struct InfiniteIter; state::Bool; end
+        Base.iterate(i::InfiniteIter, state=1) = (state, !state)
+        Base.IteratorSize(i::InfiniteIter) = Base.IsInfinite()
```

Comment:
```suggestion
        Base.IteratorSize(::Type{InfiniteIter}) = Base.IsInfinite()
```
Traits like `Base.IteratorSize` should be defined for types, not their instances.